x-pack/plugins/timelines/public/components/t_grid/shared/index.tsx

```diff
@@ -24,7 +24,16 @@ const heights = {
   short: 250,
 };
 
-export const TimelineContext = createContext<{ timelineId: string | null }>({ timelineId: null });
+export const TimelineContext = createContext<{
+  timelineId: string | null;
```

Comment:
Looks like most of this is added to context so that the row highlight effect can be achieved. I don't think it's a good idea to have some of the tgrid state in context, and some in redux, so I created a draft pr to remove this https://github.com/elastic/kibana/pull/119129 , since timelineId is nothing more than a static string to identify where in the page a timeline is, I don't think this context should have ever been added in the first place. I also think Eui is adding an easy way to do what these values stored in context are used for relatively soon https://github.com/elastic/eui/issues/4401 , and since this will cause every cell in the grid to re-render every time any of this changes, think we should hold off on adding this as it could impact performance. 
![obs_context](https://user-images.githubusercontent.com/56408403/142883820-879e6161-aec3-4735-9994-35ca09f3d218.gif)
![sec_no_context](https://user-images.githubusercontent.com/56408403/142883853-d241f245-a1d9-49cb-a62a-d72510b7e85d.gif)

Reply:
Eww, that is absolutely unacceptable. It would be nice to have native support for row selection by Eui, it would make half of this PR unnecessary.